sdk/communication/azure-communication-common/src/main/java/com/azure/communication/common/CommunicationIdentifier.java

```diff
@@ -23,27 +98,40 @@ public static CommunicationIdentifier fromRawId(String rawId) {
             throw new IllegalArgumentException("The parameter [rawId] cannot be null to empty.");
         }
 
-        if (rawId.startsWith("4:")) {
-            return new PhoneNumberIdentifier(rawId.substring("4:".length()));
+        if (rawId.startsWith(PHONE_NUMBER)) {
+            return new PhoneNumberIdentifier(rawId.substring(PHONE_NUMBER.length()));
         }
         final String[] segments = rawId.split(":");
-        if (segments.length < 3) {
+        if (segments.length != 3) {
+            if (segments.length == 2 && rawId.startsWith(BOT_GLOBAL)) {
+                return new MicrosoftBotIdentifier(segments[1], CommunicationCloudEnvironment.PUBLIC, true);
+            }
             return new UnknownIdentifier(rawId);
         }
 
         final String prefix = segments[0] + ":" + segments[1] + ":";
         final String suffix = rawId.substring(prefix.length());
 
-        if ("8:teamsvisitor:".equals(prefix)) {
+        if (TEAMS_USER_ANONYMOUS.equals(prefix)) {
```

Comment:
wdyt about replacing it with a switch block? it looks more readable for android instead of calling equals each time.

Reply:
java has switches? 😮 (just  joking 😁)

Reply:
@AikoBB the version of Java used in the pipelines doesn't support switches. I've made this change at first and had to roll back it.